rust/otap-dataflow/crates/otap/src/signal_type_router.rs

```diff
@@ -0,0 +1,234 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Signal type router processor for OTAP pipelines.
+//!
+//! Current behavior: pass-through. All signals are forwarded unchanged.
+//! Multi-port routing will be added later at the effect handler level.
+
+use crate::OTAP_PROCESSOR_FACTORIES;
+use crate::pdata::OtapPdata;
+use async_trait::async_trait;
+use linkme::distributed_slice;
+use otap_df_config::PortName;
+use otap_df_config::error::Error as ConfigError;
+use otap_df_config::experimental::SignalType;
+use otap_df_config::node::NodeUserConfig;
+use otap_df_engine::ProcessorFactory;
+use otap_df_engine::config::ProcessorConfig;
+use otap_df_engine::error::Error as EngineError;
+use otap_df_engine::local::processor as local;
+use otap_df_engine::message::Message;
+use otap_df_engine::processor::ProcessorWrapper;
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::sync::Arc;
+
+/// URN for the SignalTypeRouter processor
+pub const SIGNAL_TYPE_ROUTER_URN: &str = "urn:otap:processor:signal_type_router";
+
+/// Configuration for the SignalTypeRouter processor
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct SignalTypeRouterConfig {
+    /// Named out ports to route traces signals to (broadcast if more than one).
+    #[serde(default)]
+    pub traces_ports: Vec<PortName>,
+    /// Named out ports to route metrics signals to (broadcast if more than one).
+    #[serde(default)]
+    pub metrics_ports: Vec<PortName>,
+    /// Named out ports to route logs signals to (broadcast if more than one).
+    #[serde(default)]
+    pub logs_ports: Vec<PortName>,
+    /// Optional default out port name to use when a mapping is not defined.
+    /// If not set, the engine's single-port fallback or configured default is used.
+    #[serde(default)]
+    pub default_port: Option<PortName>,
```

Comment:
I thought the plan was to reuse the out port configuration. There is a method in the effect handler to discover the out port effectively connected.

Reply:
I was a little confused since the `SignalTypeRouter` is routing signals based on types and I do not see how the EffectHandler's `connected_ports()` would tell me which port(s) are to be used for which signal types other than establishing a `default_port`. Is there needed additional configuration in the engine or effectHandler than will more clearly define the `out_port` to use per `signal_type`?

Reply:
Ok, I removed the `*_ports` vecs and just consume the assigned or `default_port` from the engine

Reply:
I have updated as we discussed with 3 default out_ports for now as a simple approach.